### PR TITLE
[ILM] check if delete phase has actions (#65847)

### LINKED ISSUE #65847: [ILM] Delete phase can be created without delete action (empty actions object)
Currently, it's possible to create an ILM policy with a Delete phase with an empty `actions` object so that the phase doesn't contain a delete action. This will result in a policy that will not delete indices. 
How to reproduce: 
```
PUT _ilm/policy/my_policy
{
  "policy": {
    "phases": {
      "delete": {
        "actions": {}
      }
    }
  }
}
```

### PR DESCRIPTION
**Pull request by dlbas**
Fixes #65847

Added validation for the Delete phase with empty actions.

### CODE REVIEW

**Comment by astefan** [comment]
@dlbas thank you for your contribution, but it seem you didn't sign the contributor agreement.

**Comment by dlbas** [comment]
@astefan sorry, I believe I did. Is there a way to recheck my CLA signature?

**Comment by dlbas** [comment]
seems like ok now :)

**Comment by andreidan** [comment]
@elasticmachine ok to test

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ilm/LifecyclePolicy.java [hunk 1]
```diff
@@ -80,7 +80,10 @@ public LifecyclePolicy(String name, Map<String, Phase> phases) {
             if (ALLOWED_ACTIONS.containsKey(phase.getName()) == false) {
                 throw new IllegalArgumentException("Lifecycle does not support phase [" + phase.getName() + "]");
             }
-            phase.getActions().forEach((actionName, action) -> {
+           if (phase.getName().equals("delete") && phase.getActions().size() == 0) {
+               throw new IllegalArgumentException("phase [" + phase.getName() + "] must contain actions");
```
Comment (andreidan) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/ilm/LifecyclePolicy.java:84]: In order to keep the messaging around phase and actions consistent with the below validation, can you please change the message to "must define actions"?

 ```suggestion
               throw new IllegalArgumentException("phase [" + phase.getName() + "] must define actions");
```

File: client/rest-high-level/src/test/java/org/elasticsearch/client/ilm/LifecyclePolicyTests.java [hunk 2]
```diff
@@ -192,6 +192,17 @@ public void testValidateDeletePhase() {
...
 
+    public void testValidateEmptyDeletePhase() {
+        Map<String, LifecycleAction> actions = new HashMap<>();
+
+        Phase delete = new Phase("delete", TimeValue.ZERO, actions);
+        Map<String, Phase> phases = Collections.singletonMap("delete", delete);
+
+        Exception e = expectThrows(IllegalArgumentException.class,
+            () -> new LifecyclePolicy(lifecycleName, phases));
+        assertThat(e.getMessage(), equalTo("phase [" + delete.getName() + "] must contain actions"));
```
Comment (andreidan) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/ilm/LifecyclePolicyTests.java:203]:
```suggestion
        assertThat(e.getMessage(), equalTo("phase [" + delete.getName() + "] must define actions"));
```

**Review by andreidan — CHANGES_REQUESTED** [review]
Thanks for fixing this issue @dlbas 

This looks great, I've left a super minor suggestion related to the error message.

**Review by andreidan — APPROVED** [review]
LGTM, thanks for fixing this @dlbas

**Pull request merged**